### PR TITLE
Update GetDigest interface to match GetSKI

### PR DESCRIPTION
**Pull request by jacobhaven**
Now returns `ok bool` instead of `error`.

### CODE REVIEW

**Comment by jacobhaven** [comment]
@lziest Fair. I really just care about them being consistent, so I swapped GetDigest to return `"can't comute digest for non-RSA public key"` instead of `false` and reset GetSKI. 
I'll rebase before merging if this looks good.

File: server/server.go [hunk 1]
```diff
@@ -56,7 +56,7 @@ func (keys *defaultKeystore) Add(op *gokeyless.Operation, priv crypto.Signer) er
 	keys.Lock()
 	defer keys.Unlock()
 
-	if digest, ok := gokeyless.GetDigest(priv.Public()); ok {
+	if digest, err := gokeyless.GetDigest(priv.Public()); err == nil {
```
Comment (lziest) [inline server/server.go]: return the err if it's not nil?

Comment (jacobhaven) [inline server/server.go]: No, if GetDigest will only succeed for RSA keys. We should just discard the error it throws for ECDSA keys (and not have a digest registered for that key).

Comment (lziest) [inline server/server.go]: Thanks! put it in the comment?

Comment (jacobhaven) [inline server/server.go]: It's sort of implicit, just the same as all the following optional map keys. But, sure.

**Comment by jacobhaven** [comment]
*rebased and squashed commits* Looks good to merge?

**Comment by lziest** [comment]
LGTM

**Pull request merged**